unittests/internaltests.py

```diff
@@ -674,6 +674,33 @@ def _call_pkgbin(self, args, env=None):
                         for lib in ('pthread', 'm', 'c', 'dl', 'rt'):
                             self.assertNotIn(f'lib{lib}.a', link_arg, msg=link_args)
 
+    def test_program_version(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            script_path = Path(tmpdir) / 'script.py'
+            script_path.write_text('import sys\nprint(sys.argv[1])\n', encoding='utf-8')
+            script_path.chmod(0o755)
+
+            for output, expected in {
+                    '': None,
+                    '1': None,
+                    '1.2.4': '1.2.4',
+                    '1 1.2.4': '1.2.4',
+                    'foo version 1.2.4': '1.2.4',
+                    'foo 1.2.4.': '1.2.4',
+                    'foo 1.2.4': '1.2.4',
+                    'foo 1.2.4 bar': '1.2.4',
+                    '50 5.4.0': '5.4.0',
+                    'This is perl 5, version 40, subversion 0 (v5.40.0)': '5.40.0',
+                    'git version 2.48.0.rc1': '2.48.0',
+            }.items():
+                prog = ExternalProgram('script', command=[python_command, script_path, output], silent=True)
```

Comment:
Test failure on Windows is because subprocess cannot handle Path objects until python 3.8 due to a Windows-specific bug. You'll need to convert script_path to a string yourself before passing it to `command=`.

Reply:
Okay, adapted this accordingly, thanks!